src/stan-users-guide/time-series.qmd

```diff
@@ -634,334 +634,116 @@ time of the original iterations.
-up in the same way by computing sufficient statistics.
-
-### Analytic posterior {-}
-
-With the Dirichlet-multinomial HMM, the posterior can be computed analytically because the Dirichlet is the conjugate prior to the multinomial.  The following example illustrates how a Stan model can define the posterior analytically.  This is possible in the Stan language because the model only needs to define the conditional probability of the parameters given the data up to a proportion, which can be done by defining the (unnormalized) joint probability or the (unnormalized) conditional posterior, or anything in between.
-
-The model has the same data and parameters as the previous models, but
-now computes the posterior Dirichlet parameters in the transformed
-data block.
-
-```stan
-transformed data {
-  vector<lower=0>[K] alpha_post[K];
-  vector<lower=0>[V] beta_post[K];
-  for (k in 1:K) {
-    alpha_post[k] = alpha;
-  }
-  for (t in 2:T) {
-    alpha_post[z[t - 1], z[t]] += 1;
-  }
-  for (k in 1:K) {
-    beta_post[k] = beta;
-  }
-  for (t in 1:T) {
-    beta_post[z[t], w[t]] += 1;
+transformed parameters {
+  // Build transition matrix
+  matrix[3, 3] gamma;
+  for (k in 1:3) gamma[k, ] = to_row_vector(gamma_arr[k]);
+
+  // Compute the log likelihoods in each possible state
+  matrix[3, N] log_omega;
+  for (n in 1:N) {
+    for (i in 1:3) {
+      log_omega[i, n] = normal_lpdf(y[n] | mu[i], sigma);
+    }
   }
 }
-```
-
-The posterior can now be written analytically as follows.
 
-```stan
 model {
-  for (k in 1:K) {
-    theta[k] ~ dirichlet(alpha_post[k]);
-  }
-  for (k in 1:K) {
-    phi[k] ~ dirichlet(beta_post[k]);
-  }
-}
-```
-
-
-### Semisupervised estimation {-}
-
-HMMs can be estimated in a fully unsupervised fashion without any data
-for which latent states are known.  The resulting posteriors are
-typically extremely multimodal.  An intermediate solution is to use
-semisupervised estimation, which is based on a combination of
-supervised and unsupervised data.  Implementing this estimation
-strategy in Stan requires calculating the probability of an output
-sequence with an unknown state sequence.  This is a marginalization
-problem, and for HMMs, it is computed with the so-called forward
-algorithm.
-
-In Stan, the forward algorithm is coded as follows.  First, two additional data variable are declared for the unsupervised data.
+  // prior
+  mu ~ normal(0, 1);
+  sigma ~ normal(0, 1);
 
-```stan
-data {
-  // ...
-  int<lower=1> T_unsup;             // num unsupervised items
-  array[T_unsup] int<lower=1, upper=V> u;  // unsup words
-  // ...
+  // Increment target by log p(y | mu, sigma, Gamma, rho)
+  target += hmm_marginal(log_omega, gamma, rho);
 }
 ```
 
-The model for the supervised data does not change; the unsupervised
-data are handled with the following Stan implementation of the forward
-algorithm.
+The last function `hmm_marginal` takes in all the ingredients of the HMM,
```

Comment:
Eliminate the comma---English doesn't use commas between conjunctions unless there are more than two.  So it's just "A and B", but it's either "A, B, and C" (Oxford style) or "A, B and C" (defective American style).

Reply:
Ok. Although I was under the impression they didn't use the Oxford comma in the UK. I remember noticing this during StanCon in Cambridge and I figured out the Cambridge folks were just taking a piss. But when I asked some of the locals, it turned out they were familiar with the Oxford comma.